app/model/SubscriptionOps.scala

```diff
@@ -3,22 +3,36 @@ package model
 import com.github.nscala_time.time.OrderingImplicits._
 import com.gu.memsub.Benefit._
 import com.gu.memsub.BillingPeriod.{OneOffPeriod, SixWeeks}
+import com.gu.memsub.Product.{Delivery, Voucher}
 import com.gu.memsub._
 import com.gu.memsub.subsv2.SubscriptionPlan.{Paid, PaperPlan, WeeklyPlan}
 import com.gu.memsub.subsv2.{PaidCharge, PaidSubscriptionPlan, Subscription}
 import com.typesafe.scalalogging.LazyLogging
 import controllers.ContextLogging
 import model.BillingPeriodOps._
 import org.joda.time.LocalDate.now
+import PartialFunction.cond
+import scala.reflect.internal.util.StringOps
+import scalaz.syntax.std.boolean._
 
 object SubscriptionOps extends LazyLogging {
 
   type WeeklyPlanOneOff = PaidSubscriptionPlan[Product.Weekly, PaidCharge[Weekly.type, OneOffPeriod]]
 
   implicit class EnrichedPaidSubscription[P <: Paid](subscription: Subscription[P]) {
 
+    def isHomeDelivery: Boolean = containsPlanFor(Delivery)
+
+    def isVoucher: Boolean = containsPlanFor(Voucher)
+
+    def isDigitalPack: Boolean = containsPlanFor(com.gu.memsub.Product.Digipack)
+
+    def isGuardianWeekly = subscription.plans.list.exists(plan => cond(plan.product) { case _: Product.Weekly => true })
+
     private def containsPlanFor(p:Product):Boolean = subscription.plans.list.exists(_.product == p)
 
+    def hasDigitalPack: Boolean = subscription.plans.list.exists(_.charges.benefits.list.contains(Digipack))
```

Comment:
bear in mind plans includes expired ones as well, so this is more like `hasOrHadDigitalPack`